packages/ssx-sdk/src/ssx.ts

```diff
@@ -51,7 +53,12 @@ interface SignatureEncryptionConfig  extends SSXEncryptionModuleConfig {
  * Configuration for managing SSX Modules
  */
 interface SSXModuleConfig {
-  encryption: boolean | SSXEncryptionModuleConfig | IEncryption;
+  encryption: boolean | SSXEncryptionModuleConfig;
+  storage?: boolean | { [key: string]: any };
+  dataVault?: boolean | { [key: string]: any };
+  // encryption: boolean | SSXEncryptionModuleConfig | (() => IEncryption);
+  // storage?: boolean | { [key: string]: any } | (() => IStorage);
+  // dataVault?: boolean | { [key: string]: any } | (() => IDataVault);
```

Comment:
The goal here is for a user to be able to provide different configuration options. Currently going to ignore support for passing custom modules. But it would look like one of these

```js
const config = {
  siweConfig: {
    statement: "Sign up for my cool app",
  },
  modules: {
    encryption: true,
    storage: () => {
      return new CustomStorageMethod();  // commented out for now
    },
    dataVault: {
      prefix: "my-app",
    },
  },
};
```